### ISSUE TITLE
TaskInvoker.PlayAnimation throws exception as of b2310d7

### ISSUE THREAD

**Issue by nomakewan** [crosire/scripthookvdotnet#1147]
**Version**
3.6.0 latest source build (c5724ca)

**Description**
Due to the change in b2310d7 to AnimationBlendDelta in which values passed to it cannot be negative, PlayAnimation in any existing mod script that passes a negative value to BlendInSpeed or BlendOutSpeed (common considering the problem I will illustrate in a moment) will throw an exception and end the script. While this can be written around by mod maintainers, unfortunately SHVDN's own internal implementation of PlayAnimation is affected by this exception if the script in question does not pass an explicit value for BlendOutSpeed. For example, the following code when run will throw the same "The value should be positive." exception:

`Ped.Task.PlayAnimation("melee@unarmed@streamed_core_fps", MeleeAttacks.SelectRandomElement());`

This is because SHVDN, when passed only the dictionary and animation name, will translate that request to:

`PlayAnimation("melee@unarmed@streamed_core_fps", MeleeAttacks.SelectRandomElement(), 8f, -8f, -1, AnimationFlags.None, 0f);`

Where BlendOutSpeed is -8f, a negative value, which is not allowed by AnimationBlendDelta. Similarly if only a single 'speed' value is passed to PlayAnimation, it gets translated to:

`PlayAnimation(animDict, animName, speed, 0f - speed, duration, AnimationFlags.None, playbackRate);`

Where obviously either BlendInSpeed will be negative (if the user passed a negative speed value), or BlendOutSpeed will be negative if the user passed a positive speed value (due to 0f minus a positive value returning negative).

I would open a PR to resolve this problem myself but as I do not know what the intent behind this change to PlayAnimation and the addition of AnimationBlendDelta was, I do not know what the correct resolution would be according to this new implementation. My apologies.

**Comment by kagikn** [comment]
I have to ask this, does nagating blend delta values actually make difference in `TASK_PLAY_ANIM`? I haven't spot any difference yet, but ysc scripts pass a negative value as the blend out delta argument.
I made `AnimationBlendDelta` not accept negative value because I thought blend out delta was always negative.

**Comment by kagikn** [comment]
Regarding to `TASK_PLAY_ANIM`, the signs of blend delta values doesn't make any difference. 
I got confident the sign of blend delta values doesn't make any difference when I spotted `fwAnimHelpers::CalcBlendDuration(fBlendInDelta)` in `PlayAnimScriptCommand()` in part of some leaked implementation code, and the sign of them doesn't as a matter of fact. Both blend in delta and blend out delta will be stored as blend in/out durations in a `CTaskScriptedAnimation` instance (The blend in duration and blend out duration are at `CTaskScriptedAnimation + 0xC8` and `CTaskScriptedAnimation + 0xCC` respectively).
Also turned out `TASK_SYNCHRONIZED_SCENE` uses the fallback blend in value `1000.0` if blendInDelta` or `moverBlendDelta` (4th or 9th argument) is not greater than zero and uses the fallback blend in value `1000.0` if `blendOutDelta` (5th argument) is not less than zero.

I'll fix the issue where some `PlayAnimation` overloads throws exception for using negative values as the blendOutDelta value, but there's no point in `PlayAnimation` acceptting the zero value (including minus zero) as blend delta values because blending will never end. I guess we might as well make `AnimationBlendDelta` not even accept zero.
